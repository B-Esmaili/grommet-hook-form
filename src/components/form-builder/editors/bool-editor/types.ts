import { FormEditorPropsBase } from '../../types';

export type BoolEditorProps = FormEditorPropsBase & {
    defaultValue? : boolean,
    controlType : "checkbox" | "switch"
}