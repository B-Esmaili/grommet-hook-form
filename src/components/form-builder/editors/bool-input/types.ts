import { FormEditorPropsBase } from '../../types';

export type BoolInputProps = FormEditorPropsBase & {
    defaultValue? : boolean,
    controlType : "checkbox" | "switch"
}