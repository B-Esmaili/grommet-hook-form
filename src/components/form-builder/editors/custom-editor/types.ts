import { FormEditorPropsBase } from "../../../form-builder/types";
import React from "react";

export type CustomEditorProps = FormEditorPropsBase & {
  editorComponent: React.ReactElement;
  props?:any
};
