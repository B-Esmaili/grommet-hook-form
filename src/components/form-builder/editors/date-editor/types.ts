import { FormEditorPropsBase } from "../../types";
import {DateInputProps} from "grommet";

export type DateEditorProps = FormEditorPropsBase & {
    minDate?: string,
    maxDate?: string,
    dateInputProps?:DateInputProps
    defaultValue?: string | Date
}