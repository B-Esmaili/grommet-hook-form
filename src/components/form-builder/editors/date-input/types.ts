import { FormEditorPropsBase } from "../../types";

export type DateInputProps = FormEditorPropsBase & {
    minDate?: string,
    maxDate?: string,
    defaultValue: string | number
}