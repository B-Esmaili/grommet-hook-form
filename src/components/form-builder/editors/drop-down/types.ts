import { AxiosRequestConfig } from "axios";
import MockAdapter from "axios-mock-adapter/types";
import React from "react";
import { SelectProps } from "grommet";

export interface RemoteDataSource {
  request: string | AxiosRequestConfig;
  searchKey?: string;
  pageKey?: string;
  pageSizeKey?: string;
  pageSize?: number;
  extraParams?: any;
  onResponse?: (data: any, headers: any) => any;
  onRequest?: (data: any, header: any) => any;
  listKey?: string;
  totalKey?: string;
  mockResponse?: (req: MockAdapter) => void;
}

export interface OptionProps {
  label: React.ReactNode;
  selected: boolean;
}

export interface DropDownOperations {
  setValue: (setter: (value: any[] | any) => any[] | any) => void;
}

export interface DropDownProps {
  itemValueKey: string;
  itemLabelKey: string;
  labelWrap?: React.ReactElement;
  plainLabel?: boolean;
  renderItemLabel?: <TOption = any>(
    option: TOption,
    operations: DropDownOperations,
    index: number
  ) => React.ReactNode;
  multiple?: boolean;
  options: any[] | RemoteDataSource;
  placeholder?: React.ReactNode;
  onSearch?: (text: string, options: any[]) => any[];
  renderItem?: <TOption = any>(options: TOption,selected : boolean) => React.ReactNode;
  selectProps?: SelectProps;
  searchDebounce?: number;
}
