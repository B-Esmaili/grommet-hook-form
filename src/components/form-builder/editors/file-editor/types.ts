import { FileInputProps  } from "grommet";

export interface FileEditorProps {
  multiple?: boolean;
  onRemove?:(files : File[])=>void,
  fileInputProps?: Omit<
  FileInputProps,
    "onChange" | "messages" | "multiple" | "ref" | "name"
  >;
}
