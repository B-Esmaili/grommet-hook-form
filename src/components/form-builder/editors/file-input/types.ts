import { FileInputProps as GrommetFileInputProps } from "grommet";

export interface FileInputProps {
  multiple?: boolean;
  fileInputProps?: Omit<
    GrommetFileInputProps,
    "onChange" | "messages" | "multiple" | "ref" | "name"
  >;
}
