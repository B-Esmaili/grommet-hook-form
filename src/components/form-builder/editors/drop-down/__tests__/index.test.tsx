import { cleanup, render } from "@testing-library/react";
import { FormBuilder } from "../../../../form-builder";
import { FormField, FormFieldType } from "../../../../form-builder/types";
import { Box } from "grommet";
// import { FormBuilder } from "../../../../form-builder/form-builder";
// import { FormFieldType } from "../../../../form-builder/types";
// import "jest-styled-components";
// import { act } from "react-dom/test-utils";
describe("DropDown Editor", () => {
  afterEach(cleanup);

  it("custom label", () => {
    const Form = () => {
      return (
        <FormBuilder
          fields={
            [
              {
                name: "option",
                label: "option",
                type: FormFieldType.DropDown,
                defaultValue: ["opt1", "opt2"],
                options: [
                  {
                    text: "Opt1",
                    value: "opt1",
                  },
                  {
                    text: "Opt2",
                    value: "opt2",
                  },
                ],
                renderItemLabel: ({ text }: any) => {
                  return <Box>ITEM:{text}</Box>;
                },
                itemValueKey: "value",
                itemLabelKey: "text",
              },
            ] as FormField[]
          }
        />
      );
    };

    let { getAllByText } = render(<Form />);
    let elements = getAllByText(/ITEM:(Opt1|Opt2)/i);

    expect(elements.length).toBe(2);
  });
});

export {};
