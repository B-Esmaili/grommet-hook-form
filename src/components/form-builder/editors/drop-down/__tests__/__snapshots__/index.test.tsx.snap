// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Editor renders default value 1`] = `
.c15 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c15 g {
  fill: inherit;
  stroke: inherit;
}

.c15 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c15 *[stroke*="#"],
.c15 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c15 *[fill-rule],
.c15 *[FILL-RULE],
.c15 *[fill*="#"],
.c15 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: body;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: no-area;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus > circle,
.c7:focus > ellipse,
.c7:focus > line,
.c7:focus > path,
.c7:focus > polygon,
.c7:focus > polyline,
.c7:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus::-moz-focus-inner {
  border: 0;
}

.c5 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c0 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  grid-template-areas: "body" "actions";
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 96px;
}

.c12 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
}

.c12::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c12::-moz-placeholder {
  color: #AAAAAA;
}

.c12:-ms-input-placeholder {
  color: #AAAAAA;
}

.c12::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c12::-moz-focus-inner {
  border: none;
  outline: none;
}

.c12:-moz-placeholder,
.c12::-moz-placeholder {
  opacity: 1;
}

.c11 {
  position: relative;
  width: 100%;
}

.c13 {
  cursor: pointer;
}

.c8 {
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c6 {
  width: 15em;
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c14 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    margin-top: 3px;
    margin-bottom: 3px;
  }
}

<div
  class=""
>
  <form>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <div
              class="c4 "
            >
              <label
                class="c5"
              >
                <label
                  class="c6"
                >
                  test
                </label>
              </label>
              <div
                class="c3 FormField__FormFieldContentBox-m9hood-1"
              >
                <button
                  aria-label="Open Drop"
                  class="c7 c8"
                  type="button"
                >
                  <div
                    class="c9"
                  >
                    <div
                      class="c10"
                    >
                      <div
                        class="c11"
                      >
                        <input
                          autocomplete="off"
                          class="c12 c13"
                          readonly=""
                          tabindex="-1"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="c14"
                      style="min-width: auto;"
                    >
                      <svg
                        aria-label="FormDown"
                        class="c15"
                        viewBox="0 0 24 24"
                      >
                        <polyline
                          fill="none"
                          points="18 9 12 15 6 9"
                          stroke="#000"
                          stroke-width="2"
                        />
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
              <div
                class="c16"
                color="status-critical"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
